### PR TITLE
Update Angular to 19

### PR DESCRIPTION
**Pull request by nevio18324**

### CODE REVIEW

File: frontend/src/app/shared/sidepanel/sidepanel.component.html [hunk 1]
```diff
@@ -3,6 +3,7 @@
     (backdropClick)="close()"
     [cdkConnectedOverlayHasBackdrop]="true"
     [cdkConnectedOverlayHeight]="'100vh'"
+    [cdkConnectedOverlayWidth]="'800px'"
```
Comment (nevio18324) [inline frontend/src/app/shared/sidepanel/sidepanel.component.html:6]: ?

**Review by nevio18324 — COMMENTED** [review]
remove all deprication warning after `npm i`
use angular builder for jest in angular.json

File: frontend/src/app/shared/custom/dialog-template-core/dialog-template-core.component.scss [hunk 2]
```diff
@@ -1,8 +1,13 @@
-@import "custom_bootstrap";
+@use "variables";
+@use "custom_bootstrap";
+@use "bootstrap";
 
 mat-dialog-content {
   //scrollbar-gutter: stable both-edges;
   $dialog-header-height: 48px;
+  $top-bar-height: variables.$top-bar-height;
+  $dialog-content-padding-y: variables.$dialog-content-padding-y;
```
[the remark this answers is not in the document]
Comment (kcinay055679) [reply frontend/src/app/shared/custom/dialog-template-core/dialog-template-core.component.scss]: this is still not fixed.
There is an easy solution

File: frontend/src/app/shared/sidepanel/sidepanel.component.html (hunk 1 above)
Comment (MasterEvarior) [reply frontend/src/app/shared/sidepanel/sidepanel.component.html:6]: If you resolve a conversation, please provide some context as to the answer if it is not obvious. What was the question? Why is this okay, not okay, not relevant, ....

Comment (nevio18324) [reply frontend/src/app/shared/sidepanel/sidepanel.component.html:6]: the question was why is this needed now and wasn't before?
Because angular/material changed the Sidepanel it now has 100vw/100% instead of the 800px it used to have

**Comment by MasterEvarior** [comment]
Don't forget to bump all the dependencies once more before merging this.

File: frontend/src/app/shared/side-panel/side-panel.component.html [hunk 3]
```diff
@@ -3,6 +3,7 @@
     (backdropClick)="close()"
     [cdkConnectedOverlayHasBackdrop]="true"
     [cdkConnectedOverlayHeight]="'100vh'"
+    [cdkConnectedOverlayWidth]="'800px'"
```
Comment (peggimann) [inline frontend/src/app/shared/side-panel/side-panel.component.html:6]: why is this added?

File: frontend/src/style/styles.scss [hunk 4]
```diff
@@ -99,20 +100,20 @@ body.okr {
...
   @extend .key-result-attribute-show;
-  border: solid $key-result-attribute-edit-border 1px;
+  border: solid variables.$key-result-attribute-edit-border 1px;
 }
 
 .dialog-data-show {
   @extend .okr-form-label;
   @extend .linebreak;
-  @extend .bg-display-element;
+  @extend .bg-display-element !optional;
```
Comment (peggimann) [inline frontend/src/style/styles.scss:116]: is this really needed?

File: frontend/eslint.config.mjs [hunk 5]
```diff
@@ -130,6 +130,8 @@ export default tsEslint.config(
           style: 'kebab-case',
         },
       ],
+      // Off because angular our files are not standalone
```
Comment (peggimann) [inline frontend/eslint.config.mjs:133]: nice one :rofl:

**Review by peggimann — CHANGES_REQUESTED** [review]

**Review by peggimann — APPROVED** [review]
